code/modules/mechs/equipment/utility.dm

```diff
@@ -205,19 +208,106 @@
 	var/l_max_bright = 0.9
 	var/l_inner_range = 1
 	var/l_outer_range = 6
+	var/mode = MODE_CIRCULAR
+	var/obj/effect/light_effect
+	/// Defines how far the directiona light is projected
+	var/effect_steps = 2
 	origin_tech = list(TECH_MATERIAL = 1, TECH_ENGINEERING = 1)
 
+/obj/item/mech_equipment/light/Initialize()
+	light_effect = new /obj/effect(src)
+	. = ..()
+
+/obj/item/mech_equipment/light/proc/on_move(atom/movable/mover, atom/old_loc, atom/new_loc)
+	SIGNAL_HANDLER
+	if(mode == MODE_CIRCULAR || !on)
+		return
+	var/turf/current = get_turf(owner)
+	var/turf/temp = get_step(src, owner.dir)
+	for(var/i = 0, i <= effect_steps, i++ )
+		if(temp.opacity)
+			break
+		for(var/atom/thing in temp.contents)
+			if(thing.opacity)
+				goto end
```

Comment:
i dont like this change it to a flag with an if check
this is also extremely expensive and you should figure out how to change it anyway

Reply:
propose a better implementation?

Reply:
The way you have changed canopies has probably borked the code to interact with items in the mech, check that you didn't actually break this before this is merged